services/core/src/main/java/org/activiti/cloud/services/messages/core/processor/ReceiveMessagePayloadGroupProcessor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.cloud.services.messages.core.processor;
+
+import static org.activiti.cloud.services.messages.core.integration.MessageEventHeaders.MESSAGE_PAYLOAD_TYPE;
+import static org.activiti.cloud.services.messages.core.support.Predicates.MESSAGE_SENT;
+import static org.activiti.cloud.services.messages.core.support.Predicates.MESSAGE_WAITING;
+
+import java.util.Collection;
+
+import org.activiti.api.process.model.payloads.ReceiveMessagePayload;
+import org.activiti.cloud.services.messages.core.support.MessageComparators;
+import org.activiti.cloud.services.messages.core.transformer.ReceiveMessagePayloadTransformer;
+import org.springframework.integration.store.MessageGroup;
+import org.springframework.integration.store.MessageGroupStore;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+
+public class ReceiveMessagePayloadGroupProcessor extends AbstractMessageGroupProcessorHandler {
+    
+    private final MessageGroupStore messageGroupStore;
+
+    public ReceiveMessagePayloadGroupProcessor(MessageGroupStore messageGroupStore) {
+        this.messageGroupStore = messageGroupStore;
+    }
+    
+    @Override
+    protected Message<?> process(MessageGroup group) {
+        Message<?> result = group.getMessages()
+                                 .stream()
+                                 .filter(MESSAGE_SENT)
+                                 .min(MessageComparators.TIMESTAMP)
+                                 .get();
```

Comment:
This is OK, because the condition is checked in https://github.com/Activiti/activiti-cloud-messages-service/pull/1/files/5380d630c485c71b82652060c66e4af4ace197a7#diff-6d3372ac51885df2d580b682160ca5b2R69